Userland/Applications/PixelPaint/MainWidget.cpp

```diff
@@ -1093,6 +1093,7 @@ ImageEditor& MainWidget::create_new_editor(NonnullRefPtr<Image> image)
 
     image_editor.on_scale_change = [this](float scale) {
         m_zoom_combobox->set_text(String::formatted("{}%", roundf(scale * 100)));
+        current_image_editor()->update_tool_cursor();
     };
```

Comment:
Just a small thing, but I'd maybe denounce the cursor updates here, otherwise I think this might generate an excessive amount of cursors as you scroll in and out.

The snippet below prevents updating the cursor more than one per 100ms interval.

```suggestion
    auto do_cursor_update = Core::debounce([this]{
         current_image_editor()->update_tool_cursor();
    }, 100);
    image_editor.on_scale_change = [this, do_cursor_update](float scale) {
        m_zoom_combobox->set_text(String::formatted("{}%", roundf(scale * 100)));
        do_cursor_update();
    };
```
(and `#include <LibCore/Debounce.h>`)

Reply:
Hi @MacDue,
i managed ti fix that as well. Somehow i was not able to convince the compiler to accecept calling the do_cursoe_update inside the lambda -  but that was for sure my fault. Now the set_text and update cursor is done in a debounced function. I hope that's still fine.